src/main/java/com/hierynomus/smbj/share/Share.java

```diff
@@ -178,6 +178,11 @@ void closeFileId(SMB2FileId fileId) throws SMBApiException {
         sendReceive(closeReq, "Close", fileId, SUCCESS_OR_CLOSED, transactTimeout);
     }
 
+    void closeFileIdNoWait(SMB2FileId fileId) throws SMBApiException {
+        SMB2Close closeReq = new SMB2Close(dialect, sessionId, treeId, fileId);
+        sendReceive(closeReq, "Close", fileId, SUCCESS_OR_CLOSED, 0);
```

Comment:
Passing `0` to the timeout means that it will wait potentially indefinitely (it won't timeout). The method name is very misleading. Why is this needed?

Reply:
Correct, the intent was to just do send instead of sendReceive. The reasoning is that when the routers/OS close a long running connection we get a IOException and in the finally block to we close the handle. In those cases we don't care about the responses nor we want to wait for the timeout period.